### PR TITLE
refactor: Add BlockManager getters

### PR DESCRIPTION
**Pull request by maflcko**
Requested in https://github.com/bitcoin/bitcoin/pull/25781#discussion_r1061323795, but adding getters seems unrelated from removing globals, so I split it out for now.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Required for #25781 in either case

**Comment by dergoegge** [comment]
Code review ACK faf7b4f1fc35f1488567e0e4a57ecb348596b992

**Review by brunoerg — APPROVED** [review]
crACK faf7b4f1fc35f1488567e0e4a57ecb348596b992

File: src/net_processing.cpp [hunk 1]
```diff
@@ -1730,8 +1727,7 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
 
 std::optional<std::string> PeerManagerImpl::FetchBlock(NodeId peer_id, const CBlockIndex& block_index)
 {
-    if (fImporting) return "Importing...";
-    if (fReindex) return "Reindexing...";
+    if (m_chainman.m_blockman.LoadingBlocks()) return "Loading blocks ...";
```
Comment (jonatack) [inline src/net_processing.cpp:1730]: Nice refactoring.  It looks like there is also a minor user-facing behavior change here, but at first glance only affecting RPC getblockfrompeer and probably not an issue.

**Pull request merged**